tests/mgbench/workloads/base.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2022 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+from abc import ABC, abstractclassmethod
+from pathlib import Path
+
+import helpers
+from benchmark_context import BenchmarkContext
+
+
+# Base dataset class used as a template to create each individual dataset. All
+# common logic is handled here.
+class Workload(ABC):
+
+    # Name of the workload/dataset.
+    NAME = ""
+    # List of all variants of the workload/dataset that exist.
+    VARIANTS = ["default"]
+    # One of the available variants that should be used as the default variant.
+    DEFAULT_VARIANT = "default"
+
+    # List of local files that should be used to import the dataset.
+    LOCAL_FILE = None
+
+    # URLs of remote dataset files that should be used to import the dataset, compressed in gz format.
+    URL_FILE = None
+
+    # Index files
+    LOCAL_INDEX_FILE = None
+    URL_INDEX_FILE = None
```

Comment:
In the generated dataset case I think we also have to be able to inject indexes (not required to have)

Reply:
Added support for this. If you specify the workload generator, it will check if there is an index file and use it. Otherwise, you will get a log warning to add indexes in generated queries. But it will be explained in the tutorial.